samples/guide/src/main/java/okhttp3/recipes/kt/WiresharkExample.kt

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.recipes.kt
+
+import java.io.File
+import java.io.IOException
+import java.util.logging.Handler
+import java.util.logging.Level
+import java.util.logging.LogRecord
+import java.util.logging.Logger
+import javax.crypto.SecretKey
+import javax.net.ssl.SSLSession
+import javax.net.ssl.SSLSocket
+import okhttp3.Call
+import okhttp3.Connection
+import okhttp3.ConnectionSpec
+import okhttp3.ConnectionSpec.Builder
+import okhttp3.EventListener
+import okhttp3.Handshake
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.TlsVersion
+import okhttp3.TlsVersion.TLS_1_2
+import okhttp3.TlsVersion.TLS_1_3
+import okhttp3.brotli.BrotliInterceptor
+import okhttp3.dnsoverhttps.DnsOverHttps
+import okhttp3.internal.SuppressSignatureCheck
+import okio.ByteString.Companion.toByteString
+
+/**
+ * Logs SSL keys to a log file, allowing Wireshark to decode traffic and be examined with http2
+ * filter. The approach is to hook into JSSE log events for the messages between client and server
+ * during handshake, and then take the agreed masterSecret from private fields of the session.
```

Comment:
this doc is 👌

Reply:
We could probably improve with a command line

```
$ wireshark -o tls.keylog_file:/tmp/key.log -Y http2 -k
```